sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -322,21 +322,22 @@ case class StaticInvoke(
     val evaluate = if (returnNullable && !method.getReturnType.isPrimitive) {
       if (CodeGenerator.defaultValue(dataType) == "null") {
         s"""
-          ${ev.value} = $callFunc;
+          ${ev.value} = ($javaType) $callFunc;
```

Comment:
- Because some methods called by `StaticInvoke` may return a `Any`(or `Object`) value, eg:
https://github.com/apache/spark/blob/070f2bdfb968c8080de1c6614c1def978df823d4/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/json/JsonExpressionEvalUtils.scala#L103-L112

- Based on the `dataType`, obtain the `Java type` and then `cast`. 
- I think this pattern can greatly simplify the methods called by `StaticInvoke` (without writing `multiple` methods)